broker/src/main/java/org/apache/rocketmq/broker/processor/AllocateMessageQueueProcessor.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.rocketmq.common.rebalance.AllocateMessageQueueAveragely;
+import org.apache.rocketmq.common.rebalance.AllocateMessageQueueAveragelyByCircle;
+import org.apache.rocketmq.common.rebalance.AllocateMessageQueueByConfig;
+import org.apache.rocketmq.common.rebalance.AllocateMessageQueueByMachineRoom;
+import org.apache.rocketmq.common.rebalance.AllocateMessageQueueConsistentHash;
+import org.apache.rocketmq.logging.InternalLogger;
+import org.apache.rocketmq.logging.InternalLoggerFactory;
+import org.apache.rocketmq.remoting.exception.RemotingCommandException;
+import org.apache.rocketmq.remoting.netty.NettyRequestProcessor;
+import org.apache.rocketmq.remoting.protocol.RemotingCommand;
+
+public class AllocateMessageQueueProcessor implements NettyRequestProcessor {
+    private static final InternalLogger log = InternalLoggerFactory.getLogger(LoggerName.BROKER_LOGGER_NAME);
+
+    private final BrokerController brokerController;
+
+    public AllocateMessageQueueProcessor(final BrokerController brokerController) {
+        this.brokerController = brokerController;
+    }
+
+    @Override
+    public RemotingCommand processRequest(ChannelHandlerContext ctx, RemotingCommand request)
+        throws RemotingCommandException {
+        switch (request.getCode()) {
+            case RequestCode.ALLOCATE_MESSAGE_QUEUE:
+                return this.allocateMessageQueue(ctx, request);
+            default:
+                break;
+        }
+        return null;
+    }
+
+    private RemotingCommand allocateMessageQueue(ChannelHandlerContext ctx, RemotingCommand request)
+        throws RemotingCommandException {
+        final RemotingCommand response =
+            RemotingCommand.createResponseCommand(AllocateMessageQueueResponseHeader.class);
+        final AllocateMessageQueueRequestHeader requestHeader =
+            (AllocateMessageQueueRequestHeader) request.decodeCommandCustomHeader(AllocateMessageQueueRequestHeader.class);
+        final AllocateMessageQueueRequestBody requestBody = AllocateMessageQueueRequestBody.decode(request.getBody(),
+            AllocateMessageQueueRequestBody.class);
+
+        AllocateMessageQueueStrategy strategy = null;
+        String strategyName = requestHeader.getStrategyName();
+        if (strategyName.startsWith("MACHINE_ROOM_NEARBY")) {
+            strategy = new AllocateMachineRoomNearby(new AllocateMessageQueueAveragely(),
+                new AllocateMachineRoomNearby.MachineRoomResolver() {
+                    @Override
+                    public String brokerDeployIn(MessageQueue messageQueue) {
+                        return messageQueue.getBrokerName().split("-")[0];
+                    }
+
+                    @Override
+                    public String consumerDeployIn(String clientID) {
+                        return clientID.split("-")[0];
+                    }
+                });
+        } else {
+            switch (strategyName) {
+                case "AVG":
```

Comment:
Enum or constant would be better，magic numbers not a good choice.

Reply:
Updated. The strings have been changed to constants.